ksql-version-metrics-client/src/test/java/io/confluent/ksql/version/metrics/collector/BasicCollectorTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.version.metrics.collector;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+import java.util.EnumSet;
+
+import io.confluent.ksql.version.metrics.KsqlVersionMetrics;
+import io.confluent.support.metrics.common.Version;
+
+@RunWith(Parameterized.class)
+public class BasicCollectorTest {
+
+  @Parameterized.Parameters
+  public static Collection<KsqlModuleType> data() {
+    return EnumSet.allOf(KsqlModuleType.class);
+  }
+
+  @Parameterized.Parameter
+  public KsqlModuleType moduleType;
+
+  @Test
+  public void testGetCollector(){
+    BasicCollector basicCollector = new BasicCollector(moduleType);
```

Comment:
Can we pass in a `TimeUtils` with a mock clock to `BasicCollector`? Then i think we should be able to do something like
```
metrics = new KsqlVersionMetrics();
metrics.setTimestamp(time);
metrics.setConfluentPlaftormVersion(Version.getVersion());
metrics.setKsqlComponentType(moduleType.name());
assertThat(basicCollector.collectMetrics(), equalTo(metrics)):
```